### ISSUE TITLE
Add check for disabledPackageSources (in NuGet.config) to Xenko Launcher

### ISSUE THREAD

**Issue by pikachuk** [xenko3d/xenko#287]
**Release Type**: Official Release

**Version**: Xenko Launcher 3.0.1

**Platform(s)**: Windows

**Describe the bug**
Because Xenko Launcher doesn't check for disablePackageSources in NuGet.conf, if the xenko nuget packages are disabled, the xenko launcher will think it's offline and won't be able to connect to the internet to download the xenko builds from the internet

**To Reproduce**
1. open NuGet.config in %APPDATA%\NuGet\NuGet.config
2. add these lines in configuration
```
<disabledPackageSources>
    <add key="Xenko" value="true" />
    <add key="Xenko Dev" value="true" />
  </disabledPackageSources>
```
3. The Xenko launcher cannot connect and download xenko builds

**Expected behavior**
The launcher should check if the xenko package are disabled and automatically enable them, or ask the user the authorization to enable them

**Screenshots**
![unknown](https://user-images.githubusercontent.com/16335601/49699406-fb23d900-fbd0-11e8-85fb-5778be0eb52d.png)

**Comment by Eideren** [comment]
Isn't this expected behavior though ? If users explicitly disable nugget sources they probably don't want xenko/stride to attempt fetching them ? I'll mark this one as enhancement instead, seeing as we could extend the window to mention why it failed.

**Comment by Kryptos-FR** [comment]
@Eideren Yes I would also expect that the Launcher doesn't find them if the sources are disabled.

Not sure if the API let's us find out about disabled packages. If so, displaying a more explicit error message could be an option.

**Comment by kalyan-4974** [comment]
Hi @stride3d/stride-contributors, I'd like to take this on for US$30 under your contribution program.

My patch adds a clear explanation when no NuGet package sources are enabled, while preserving the user's configuration. All 11 launcher tests passed, including five new cases covering empty, disabled, enabled, and mixed source configurations.

Could you confirm whether this scope works and reserve the task for me?

<details>
<summary>Validation details</summary>

Based on cc08fcee71127aebf5761f36995d10bc48cb8839. Tests use temporary configurations and empty local feeds. They also check unchanged configuration, repeated offline refreshes, and recovery to online state.

`dotnet test sources/launcher/Stride.Launcher.Tests/Stride.Launcher.Tests.csproj --nologo --verbosity minimal -m:2 -p:StrideNativeOutputName= -p:GeneratePackageOnBuild=false -p:StrideSentryDsn=false --no-restore`

The normal build couldn't find Windows native-link libraries, so the managed test run disabled native crash reporting. Assembly-processor binaries were absent and produced warnings. Manual launcher UI testing and a full release build are still outstanding.

</details>

Thanks,
Kalyan
